anndata/_core/merge.py

```diff
@@ -364,6 +380,24 @@ def _apply_to_df(self, el: pd.DataFrame, *, axis, fill_value=None):
             fill_value = np.NaN
         return el.reindex(self.new_idx, axis=axis, fill_value=fill_value)
 
+    def _apply_to_dask_array(self, el: DaskArray, *, axis, fill_value=None):
+        import dask.array as da
+
+        if fill_value is None:
+            fill_value = default_fill_value([el])
+        shape = list(el.shape)
+        if el.shape[axis] == 0:
+            # Presumably faster since it won't allocate the full array
+            shape[axis] = len(self.new_idx)
+            return da.broadcast_to(fill_value, tuple(shape))
+
+        indexer = self.old_idx.get_indexer(self.new_idx)
+
+        # Indexes real fast, and does outer indexing
+        # TODO this
+
+        return _subset(el, make_slice(indexer, axis, len(shape)))
+
```

Comment:
```suggestion
        sub_el = _subset(el, make_slice(indexer, axis, len(shape)))
        sub_el[make_slice(indexer == -1, axis, len(shape))] = fill_value
        return sub_el

```

I think this will fix it.

The indexer being generated by `self.old_idx.get_indexer(self.new_idx)` used `-1` to represent missing values. Pandas expects this, but numpy does not. By numpy indexing conventions, this just meant to grab the last value.

I bet the solution I've proposed isn't particularly efficient, but it should work.

Reply:
Great it works!